### ISSUE TITLE
Question/Feature request: Enhance initialFieldState in formReducer

### ISSUE THREAD

**Issue by asvetliakov** [davidkpiano/react-redux-form#145]
I want to add few properties to field form reducer (in my case i need to add status icon for field), what's the best way to do this?
The trick:
```
const formReducer = formReducer("user", initialFormState);

function reducer(state, action) {
  const newState = formReducer(state, action);
  switch MY_ACTION:
    return { ...newState, fields: { email: { iconStatus: "success" } } };
 ....
}
```
will work partially since there is no way to provide initial state for form or fields itself. What if i want "loading" iconStatus in fields by default? I can dispatch action in componentDidMount() of course but i think it would be good if there will be third parameter or merger function or anything else to enhance initialForm/initialFieldState

**Comment by davidkpiano** [comment]
I think this is a good idea - essentially we just need an API for decorating form reducers with the same semantics as the form reducer itself. I'll try to think of something but I'm definitely open to suggestions.

**Comment by davidkpiano** [comment]
@asvetliakov What do you think of this approach?

```js
import { formReducer, enhanceForm, updateField } from 'react-redux-form';

const userFormReducer = formReducer('user', initialState);

const fieldIconStatus = (formState, action) => {
  switch (action.type) {
    case MY_ACTION:
      return updateField(formState, { iconStatus: action.status });
    default:
      return formState;
  }
};

export default enhanceForm(userFormReducer, [
  fieldIconStatus,
  // ... any other enhancers
]);
```

**Comment by asvetliakov** [comment]
Looks very good. This way i can throw away trick with calling reducer inside reducer
What about initial form state?

**Comment by asvetliakov** [comment]
Will be result of [enhancers] merged on first run?
i.e. something like this:
```
const fieldIconStatus = (formState = { fields: { myField: { iconStatus: "disabled" } } }, action) => {
...
}
const reducer = enhanceForm(userFormReducer, [fieldIconStatus]);
const state  = reducer(undefined, {});
// state now:
{
  ....
  fields: {
    myField: {
      ...
      initialValue: '',
      iconStatus: "disabled"
    }
  }
}
```
The problem is to not add additional actions processing to the formReducer (calling reducer inside reducer is ugly but works flawlessly, and anyway it's something need to be done if you want to keep whole state tree as Immutable), the problem is getting own initial state to be merged into form initial state

**Comment by davidkpiano** [comment]
How about instead of initial state you think of it as default state?

Alternatively, I can also make an option available:

```js
formReducer('user', {
  initialFieldState: { iconStatus: 'disabled' }
});
```

**Comment by asvetliakov** [comment]
>How about instead of initial state you think of it as default state?

I.e assume that ```myField.iconState === undefined``` by default? That may work for one situation and may not work for other. I.e. this doesn't solve the problem if you want other value rather than undefined.
Second form ```initialFieldState``` is fine but has drawbacks:
* need to have two options for form state and field state if you want to modify attach values to form rather than fields
* this doesn't look very good:
```js
formReducer('user', {
  email: '', // these fields define field state too
  firstName: '',
  initialFieldState: { iconStatus: 'disabled' }, // ??? another field state?
  initialFormState: { formProp: false }
}
```
somethin like it will be more clear from consumer perspective but will break current API:
```js
formReducer('user', {
 initialValues: {
   firstName: '',
   email: ''
 },
 initialFieldState: { iconStatus: 'disabled' }
...
}
```

**Comment by davidkpiano** [comment]
Ah, sorry, meant to make that as a third option, as in:

###`formReducer(model, initialState, [options = {}])`

So, it would look like this:

```js
formReducer('user', initialState, {
  initialFieldState: { iconStatus: 'disabled' }
})
```

**Comment by asvetliakov** [comment]
This better, but why you can't agree with just merging third param into initial state?

**Comment by davidkpiano** [comment]
Just because if we need more options in the future, that would necessitate more arguments, and I wasn't to avoid the parity of the functions being too high.

**Comment by davidkpiano** [comment]
@asvetliakov Just implemented this in the `v1` branch for the upcoming version 1.0. The syntax will look like this:

```js
formReducer('user', initialState, {
  plugins: [ statusPlugin ],
  initialFieldState: { iconStatus: 'disabled' }
})
```

where `statusPlugin` can look something like this:

```js
import updateField from 'react-redux-form/utils/updateField';

export default function statusPlugin(state, action, localPath) {
  if (action.type === 'UPDATE_STATUS') {
    return updateField(state, localPath, { iconStatus: action.status });
  }

  return state;
}
```

Specifics: a `plugin` is like a normal reducer, except it's given a third argument, `localPath`, which is the path to the field for your form. So if you have a `formReducer('test', ...)`, and an action such as `actions.change('test.foo.bar', ...)`, the `localPath` will be `['foo', 'bar']` for that reducer.

**Comment by davidkpiano** [comment]
Available in latest beta: `1.0.0-beta.7`
